node/overseer/src/lib.rs

```diff
@@ -1421,490 +533,118 @@ where
 					// as well as the amount of messages from external sources to the overseer
-					// into one to_overseer value.
-					metronome_metrics.channel_fill_level_snapshot(subsystem_meters);
+					// into one `to_overseer` value.
+					metronome_metrics.channel_fill_level_snapshot(
+						subsystem_meters.iter()
+							.cloned()
+							.filter_map(|x| x)
+							.map(|(name, ref meters)| (name, meters.read()))
+					);
 
 					async move {
 						()
 					}
 				});
-			s.spawn("metrics_metronome", Box::pin(metronome));
+			overseer.spawner().spawn("metrics_metronome", Box::pin(metronome));
 		}
 
-		let this = Self {
-			subsystems,
-			s,
-			running_subsystems,
-			to_overseer_rx: to_overseer_rx.fuse(),
-			events_rx,
-			activation_external_listeners,
-			leaves,
-			active_leaves,
-			metrics,
-			span_per_active_leaf: Default::default(),
-			known_leaves: LruCache::new(KNOWN_LEAVES_CACHE_SIZE),
-			supports_parachains,
-		};
-
-		Ok((this, handler))
+		Ok((overseer, Handler(handler)))
 	}
 
-	// Stop the overseer.
+	/// Stop the overseer.
 	async fn stop(mut self) {
-		let _ = self.subsystems.candidate_validation.send_signal(OverseerSignal::Conclude).await;
-		let _ = self.subsystems.candidate_backing.send_signal(OverseerSignal::Conclude).await;
-		let _ = self.subsystems.statement_distribution.send_signal(OverseerSignal::Conclude).await;
-		let _ = self.subsystems.availability_distribution.send_signal(OverseerSignal::Conclude).await;
-		let _ = self.subsystems.availability_recovery.send_signal(OverseerSignal::Conclude).await;
-		let _ = self.subsystems.bitfield_signing.send_signal(OverseerSignal::Conclude).await;
-		let _ = self.subsystems.bitfield_distribution.send_signal(OverseerSignal::Conclude).await;
-		let _ = self.subsystems.provisioner.send_signal(OverseerSignal::Conclude).await;
-		let _ = self.subsystems.runtime_api.send_signal(OverseerSignal::Conclude).await;
-		let _ = self.subsystems.availability_store.send_signal(OverseerSignal::Conclude).await;
-		let _ = self.subsystems.network_bridge.send_signal(OverseerSignal::Conclude).await;
-		let _ = self.subsystems.chain_api.send_signal(OverseerSignal::Conclude).await;
-		let _ = self.subsystems.collator_protocol.send_signal(OverseerSignal::Conclude).await;
-		let _ = self.subsystems.collation_generation.send_signal(OverseerSignal::Conclude).await;
-		let _ = self.subsystems.approval_distribution.send_signal(OverseerSignal::Conclude).await;
-		let _ = self.subsystems.approval_voting.send_signal(OverseerSignal::Conclude).await;
-		let _ = self.subsystems.gossip_support.send_signal(OverseerSignal::Conclude).await;
-
-		let mut stop_delay = Delay::new(Duration::from_secs(STOP_DELAY)).fuse();
-
-		loop {
-			select! {
-				_ = self.running_subsystems.next() => {
-					if self.running_subsystems.is_empty() {
-						break;
-					}
-				},
-				_ = stop_delay => break,
-				complete => break,
-			}
-		}
+		::std::mem::drop(
```

Comment:
what's mem::drop for?

Reply:
Moving to a  `let _ = ..` for sake of idiomatics, I don't quite remember why I did it this way.